src/managers/TileManager.ts

```diff
@@ -38,16 +38,72 @@ class TileManager {
                 ),
                 session.seed,
             );
-            this.list.forEach((tile: Tile) => {
-                const state: TileState = TileState.Unused; // TODO
+
+            for (let i = 0; i < this.list.length; i++) {
+                const tile: Tile = this.list[i];
                 const dto: Nullable<TileDTO> =
                     dtos.find((dto: TileDTO) => dto.id === tile.id) || null;
+
+                let state: TileState = this.identifyTiles(
+                    dto,
+                    session.turnPlayer!,
+                    i,
+                );
                 tile.apply(state, dto);
-            });
+            }
+            this.getTilesInHand(); //TODO where to save/use
             this.onSync.emit();
         });
     }
 
+    private identifyTiles(
+        dto: Nullable<TileDTO>,
+        turnPlayer: int,
+        i: int,
+    ): TileState {
+        if (dto !== null) {
+            return TileState.Placed;
+        }
+        if (
+            i <
+            turnPlayer +
+                this.getStartOfGameTilesAmount(GeneralManager.getPlayers.length)
+        ) {
+            return TileState.Drawn;
+        }
+        return TileState.Unused;
+    }
+
+    private getTilesInHand(): TileDTO[] {
+        const playerAmount: int = GeneralManager.getPlayers.length;
+        const orderIndex: int = PlayerManager.getMe()!.orderIndex!;
+        const allTiles: TileDTO[] = GeneralManager.getTiles()!;
+        const tilesInHand: TileDTO[] = [];
+
+        for (let i = 0; i < allTiles.length; i++) {
+            if (
+                allTiles![i].state === TileState.Drawn &&
+                i % playerAmount === orderIndex
+            ) {
+                tilesInHand.push(allTiles[i]);
+            }
+        }
+
+        return tilesInHand;
+    }
+
+    private getStartOfGameTilesAmount(playerAmount: int): int {
```

Comment:
Would suggest renaming the method to "getInitialTileAmount" and the parameter to "playerCount" for consistency with your comment on my PR.

Reply:
or maybe even "getInitialAmount". its absolutely ok and wide spread to leave away things like "tile" when you are inside a class named "tile".